tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15716,6 +15749,56 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, llvm::StringRef StageName,
                diag::err_hlsl_maxrecordssharedwith_references_invalid_arg);
       }
     }
+
+    // Make sure NodeTrackRWInputSharing attribute cannot be applied to
+    // Input Records that are not RWDispatchNodeInputRecord
+    ArgIdx = 0;
+    while (ArgIdx < FD->getNumParams()) {
```

Comment:
This is already inside the argument loop.  Why are we iterating through all the arguments for each argument here?  I believe you just want to check the current argument.  Take a look at how `ParamTy` is used, which is only set for the current argument in the outer loop.  The inner loop code doesn't make sense based on this.